### ISSUE TITLE
New rule idea: warn on dropping a not null constraint

### ISSUE THREAD

**Issue by andrewsmith** [sbdchd/squawk#282]
I had a colleague reach out with an idea that might make a good Squawk rule. Basically, we had a column with a not null constraint that got relaxed into a nullable column via:

```sql
ALTER TABLE foo ALTER COLUMN x DROP NOT NULL;
```

This caused issues with downstream processing where other code was not expecting null values. This seemed like a "stop and think about the implications of this before proceeding" moment, akin to dropping a column or other backwards-incompatible schema changes. Would Squawk accept such a rule? I'm happy to implement it.

**Comment by chdsbd** [comment]
Sounds like a great addition! I’d definitely merge a PR for this
